searx/webapp.py

```diff
@@ -686,6 +691,9 @@ def search():
     if output_format not in ['html', 'csv', 'json', 'rss']:
         output_format = 'html'
 
+    if output_format not in get_value(settings, 'search', 'formats', default=[]):
```

Comment:
`default=[]` --> `default=['html', 'csv', 'json', 'rss']` otherwise it will break existing setup (without `searx.formats`).

Looking at the code above:

```python
OUTPUT_FORMATS = ['html', 'csv', 'json', 'rss']

...
    allowed_formats = get_value(settings, 'search', 'formats', default=OUTPUT_FORMATS)
    output_format = request.form.get('format', allowed_formats[0])
    if output_format not in OUTPUT_FORMATS:
        # unknown format: fallback to the first allowed format
        # default setting: 'html' 
        output_format  = allowed_formats[0]
    elif output_format not in allowed_formats:
        # the format is valid but not allowed: error 403
        flask.abort(403)
```

---

The fallback part keeps the API compatibility (`if output_format not in OUTPUT_FORMATS`)

Reply:
> otherwise it will break existing setup (without search.formats).

good idea to set the default including all formats .. but should we set the default in the code above or is it better placed nearby loading the settings?

Next thought: is it good to have defaults in settings.yml and other defauts in the code (e.g. nearby loading settings).

What I want is a single point of definitions .. ATM this single point is the settings,yml .. should we really add a new place where defaults could be located? (hope I was clear)

Reply:
> placed nearby loading the settings

I don't understand the location.

A simple solution for now: a the top of webapp.py (?)

---

With the searx PR 2501:
* the default value is in `searx/settings_defaults.py`
* `allowed_formats = get_value(settings, 'search', 'formats', default=OUTPUT_FORMATS)` --> `allowed_formats = settings['search']['formats']` (we are sure there is a value)

As discuss in this PR, a Settings class would be better, then the Preferences can use a Settings instance.

Reply:
> I don't understand the location.

I mean a file like ..

> With the searx PR 2501:
>    the default value is in searx/settings_defaults.py

I think a solution like PR2501 will help ... 

> As discuss in this PR, a Settings class would be better, then the Preferences can use a Settings instance.

PR2501 also implements static_path handling .. for me it is a bit hard to merge all this into tthe current searxng without to fail ... If you have time, do you like to merge this PR?  If so, I would vote to implement a class.